win32ss/user/ntuser/ghost.c

```diff
@@ -1,18 +1,100 @@
 /*
  * PROJECT:     ReactOS user32.dll
  * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
- * PURPOSE:     Ghost window handling
+ * PURPOSE:     Window ghosting feature
  * COPYRIGHT:   Copyright 2018 Katayama Hirofumi MZ (katayama.hirofumi.mz@gmail.com)
  */
 
 #include <win32k.h>
+#include "ghostwnd.h"
 
-BOOL FASTCALL IntGoGhost(PWND Window, BOOL bGo)
+#define NDEBUG
+#include <debug.h>
+
+static UNICODE_STRING GhostClass = RTL_CONSTANT_STRING(GHOSTCLASSNAME);
+static UNICODE_STRING GhostProp = RTL_CONSTANT_STRING(GHOST_PROP);
+
+BOOL FASTCALL IntIsGhostWindow(HWND hWnd)
+{
+    BOOLEAN Ret = FALSE;
+    UNICODE_STRING ClassName;
+    INT Len;
+
+    // check class name
+    RtlInitUnicodeString(&ClassName, NULL);
+    Len = NtUserGetClassName(hWnd, TRUE, &ClassName);
+    if (Len)
+    {
+        Ret = RtlEqualUnicodeString(&ClassName, &GhostClass, TRUE);
+    }
+    RtlFreeUnicodeString(&ClassName);
+
+    return Ret;
+}
+
+HWND APIENTRY NtUserGhostWindowFromHungWindow(HWND hwndHung)
```

Comment:
That's a Vista+ API.

Reply:
As long as it is not exported (yet) it's fine.